### PR TITLE
ARROW-4897: [Rust] [DataFusion] Improve rustdocs

### PR DESCRIPTION
**Pull request by andygrove**
Also removes dead code and changes the visibility on some structs.

### CODE REVIEW

File: rust/datafusion/src/execution/projection.rs [hunk 1]
```diff
@@ -15,7 +15,9 @@
 // specific language governing permissions and limitations
 // under the License.
 
-//! Execution of a projection
+//! Execution of a projection. A projection determines which columns or expressions are returned
```
Comment (paddyhoran) [inline rust/datafusion/src/execution/projection.rs]: "Execution of a projection." does not seem to fit with me as the opening documentation of this module.  Maybe something like "Defines the *projection* type relation and how it is executed."

Note: this is completely subjective...

Comment (paddyhoran) [inline rust/datafusion/src/execution/projection.rs]: "Execution of a projection." does not seem to fit with me as the opening documentation of this module.  Maybe something like "Defines the *projection* type relation and how it is executed."

Note: this is completely subjective...

File: rust/datafusion/src/execution/relation.rs [hunk 2]
```diff
@@ -15,6 +15,10 @@
 // specific language governing permissions and limitations
 // under the License.
 
+//! Trait for Relation. A relation is a representation of a set of tuples. A database table is a
```
Comment (paddyhoran) [inline rust/datafusion/src/execution/relation.rs]: Again here for me "Trait for Relation." is not really an overview of the module.  Maybe "Defines general relation related types and traits."

Note: this is completely subjective...

**Review by paddyhoran — APPROVED** [review]
+1 LGTM, few nits.  Feel free to ignore if you disagree.

**Comment by andygrove** [comment]
@paddyhoran Good feedback. Thanks. I updated.

**Review by paddyhoran — APPROVED** [review]
+1 LGTM

**Pull request closed without merging**